api/src/extensions/lib/get-extensions-settings.ts

```diff
@@ -28,11 +28,44 @@ export const getExtensionsSettings = async ({
 	const existingSettings = await service.extensionsItemService.readByQuery({ limit: -1 });
 
 	const newSettings: ExtensionSettings[] = [];
+	const removedSettingIds: string[] = [];
 
 	const localSettings = existingSettings.filter(({ source }) => source === 'local');
 	const registrySettings = existingSettings.filter(({ source }) => source === 'registry');
 	const moduleSettings = existingSettings.filter(({ source }) => source === 'module');
 
+	const updateBundleEntriesSettings = (
+		extensionSettings: ExtensionSettings,
+		extension: Extension,
+		settings: ExtensionSettings[],
```

Comment:
Would be awesome if we can find more expressive names for the params 🙏 E.g.
```suggestion
		bundle: BundleExtension,
		bundleSettings: ExtensionSettings,
		allSettings: ExtensionSettings[],
```